portal/models/questionnaire_response.py

```diff
@@ -225,7 +225,7 @@ def document_answered(self):
             for answer in combined_answers:
 
                 # Add text answer before coded answer
-                if answer.keys()[0] == 'valueCoding':
+                if list(answer.keys())[0] == 'valueCoding':
```

Comment:
```suggestion
                if next(iter(answer.keys())) == 'valueCoding':
```


A more general (more suitable to iterators) approach might be to use `next()` to get the first value of it as an iterator
```
>>> answer = {'valueCoding': {'code':[]}}
>>> hasattr(answer.keys(), '__iter__')
True
>>> next(iter(answer.keys()))
'valueCoding'
```